### PR TITLE
Feat: Implement Constructor with Ownable Initialization and Pause/Unpause Functions

### LINKED ISSUE #6: [CONTRACT] Implement Constructor with Ownable Initialization and Pause/Unpause Functions
## **Description:**  
Define the contract constructor in the `trajectfi.cairo` file to initialize the `Ownable` component, ensuring that only the deployer has major administrative control. Additionally, implement functions to pause and unpause contract operations using the `Pausable` component, restricting certain functionalities when paused. 

### **Proposed Actions:**  

1. **Constructor Implementation:**  
   - Initialize the contract with the `Ownable` pattern.  
   - Ensure the deployer is set as the initial owner via `self.ownable.initializer(owner)`  function.  

2. **Pause/Unpause Functions:**  
   - Implement a `pause()` function:  
     - Only callable by the owner.  
     - Sets a `paused` state variable to `true` for the Pausable component using the private `pause` function.  
     - Emits a `Paused` event.  
   - Implement an `unpause()` function:  
     - Only callable by the owner.  
     - Sets `paused` to `false` or the Pausable component using the private `unpause` function.   
     - Emits an `Unpaused` event.  

### **Acceptance Criteria:**  
- Contract correctly initializes with the deployer as the owner.  
- Only the owner can pause and unpause the contract.  
- Proper event emission for state changes. 

### **Additional Context:**
Follow the contibution guidelines

### PR DESCRIPTION
**Pull request by Verifieddanny**
## Description

Implemented contract constructor initialization for Ownable and Pausable components in the Trajectfi contract. The changes ensure:
- Deployer is set as the initial owner via Ownable component
- Proper pause/unpause functionality with owner-only access
- Event emission for state changes
- Maintained existing access control functionality

## Type of Change

- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [x] Contract optimization
- [ ] Documentation update
- [ ] Configuration change
- [ ] NFT collateralization flow change
- [ ] Loan terms or parameters modification

### Development Best Practices

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [x] New and existing tests pass locally with my changes

### Security Considerations

- [x] I have reviewed potential attack vectors introduced by these changes
- [x] My code handles edge cases and ensures no funds can be lost
- [ ] Any updates to loan parameters or collateralization ratios have been thoroughly tested
- [x] All state modifications are properly validated and authorized
- [x] No centralization risks are introduced

### Contract Specific Checks

- [x] Contract changes are minimal and focused on the requirements
- [x] Cairo-specific optimizations are implemented where possible
- [x] Gas usage has been analyzed and optimized
- [x] Contract interfaces maintain backwards compatibility (unless breaking change is intended)
- [x] Starknet-specific considerations have been addressed

### Testing

- [x] Unit tests cover the new functionality
- [x] Integration tests verify the changes work with other components
- [ ] Loan/collateral edge cases are tested (e.g., default scenarios, renegotiation)
- [ ] I have tested with realistic NFT values and token amounts

## Implementation Notes

- Used OpenZeppelin's Cairo implementations for Ownable and Pausable patterns
- Maintained existing AccessControl functionality while adding new features
- Implemented minimal interface changes to maintain backward compatibility
- Added proper access control checks for pause/unpause functions
- Followed Starknet best practices for event emission

## Related Issues

- Fixes #6

## Environment & Deployment

- Starknet Network: Testnet
- Cairo Version: 2.11.2
- Compiler Settings: Default optimization level

### CODE REVIEW

File: src/trajectfi.cairo [hunk 1]
```diff
@@ -104,8 +104,35 @@ pub mod Trajectfi {
...
         self.accesscontrol._grant_role(OWNER_ROLE, owner);
         self.accesscontrol._grant_role(ADMIN_ROLE, owner);
+        
+        // Initialize Pausable component
+        self.pausable.initializer();
     }
+
+      // Pause function - only callable by owner
+      #[abi(embed_v0)]
+      fn pause(ref self: ContractState) {
```
Comment (jaykayudo) [inline src/trajectfi.cairo]: Define the `pause` and `unpause` in an interface and implement the interface in this contract.
The interface should be in the `itrajectfi.cairo` file.

**Comment by jaykayudo** [comment]
Hello @Verifieddanny 
Please fix the failing CI.

**Review by jaykayudo — APPROVED** [review]
